custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js

```diff
@@ -86,10 +95,22 @@ function MapOrSectorController($scope, $compile, $location, storageService, loca
         });
     }
 
+    vm.getTopForPopup = function (event, divId) {
+        // find view height and look if (suggested top + popup height) is more. if it is higher than view height,
+        // find the difference and move popup up by that value else go with suggested top value.
+        var viewHeight = $(window).height();
+        var suggestedTop = event.layerY;
+        var popupHeight = $('#' + divId).outerHeight(true);
+        var outOfScreenHeight = suggestedTop + popupHeight - viewHeight;
+        return (viewHeight < (suggestedTop + popupHeight)) ? (suggestedTop - outOfScreenHeight) : suggestedTop;
```

Comment:
nit: a clearer condition, if I'm understanding this, might be `outOfScreenHeight > 0`